src/middlewared/middlewared/plugins/account_/privilege.py

```diff
@@ -0,0 +1,236 @@
+        )
+
+        return await self.get_instance(id)
+
+    @accepts(
+        Int("id", required=True),
+        Patch(
+            "privilege_entry",
+            "privilege_update",
+            ("rm", {"name": "builtin_name"}),
+            ("attr", {"update": True}),
+        )
+    )
+    async def do_update(self, id, data):
+        """
+        Update the privilege `id`.
+        """
+        old = await self.get_instance(id)
+        new = old.copy()
+        new["local_groups"] = [g["gid"] for g in new["local_groups"]]
+        new["ds_groups"] = [g["gid"] for g in new["ds_groups"]]
+        new.update(data)
+
+        verrors = ValidationErrors()
+
+        if new["builtin_name"]:
+            for k in ["name", "allowlist"]:
+                if new[k] != old[k]:
+                    verrors.add(f"privilege_update.{k}", "This field is read-only for built-in privileges")
+
+            builtin_privilege = BuiltinPrivileges(new["builtin_name"])
+
+            if builtin_privilege == BuiltinPrivileges.LOCAL_ADMINISTRATOR:
+                if not await self.middleware.call("group.has_password_enabled_user", new["local_groups"]):
+                    verrors.add(
+                        "privilege_update.local_groups",
+                        "None of the members of these groups has password login enabled. At least one grantee of "
+                        "the \"Local Administrator\" privilege must have password login enabled."
+                    )
+
+        verrors.check()
+
+        new.update(data)
+
+        await self._validate("privilege_update", new, id)
+
+        await self.middleware.call(
+            "datastore.update",
+            self._config.datastore,
+            id,
+            new,
+        )
+
+        return await self.get_instance(id)
+
+    @accepts(
+        Int("id")
+    )
+    async def do_delete(self, id):
+        """
+        Delete the privilege `id`.
+        """
+        privilege = await self.get_instance(id)
+        if privilege["builtin_name"]:
+            raise CallError("Unable to delete built-in privilege", errno.EPERM)
+
+        response = await self.middleware.call(
+            "datastore.delete",
+            self._config.datastore,
+            id
+        )
+
+        return response
+
+    async def _validate(self, schema_name, data, id=None):
+        verrors = ValidationErrors()
+
+        await self._ensure_unique(verrors, schema_name, "name", data["name"], id)
+
+        groups = await self._groups()
+        for i, local_group_id in enumerate(data["local_groups"]):
+            if not self._local_groups(groups, [local_group_id]):
+                verrors.add(f"{schema_name}.local_groups.{i}", "This local group does not exist")
+        for i, ds_group_id in enumerate(data["ds_groups"]):
+            if not await self._ds_groups(groups, [ds_group_id]):
+                verrors.add(f"{schema_name}.local_groups.{i}", "This Directory Service group does not exist")
```

Comment:
`   verrors.add(f"{schema_name}.local_groups.{i}", "This Directory Service group does not exist")`
Shouldn't that be `f"{schema_name}.ds_groups.{i}"`?